File: server/java/render/src/main/java/com/metaobjects/render/prompt/PromptOverrides.java
package com.metaobjects.render.prompt;

import java.util.Map;

/** Render-time overrides of the metadata defaults. style null = keep spec's; maps override per field name. */
public record PromptOverrides(PromptStyle style, Map<String, String> examples, Map<String, String> instructions) {
    public PromptOverrides {
        examples = examples == null ? Map.of() : Map.copyOf(examples);
        instructions = instructions == null ? Map.of() : Map.copyOf(instructions);
    }
    public static PromptOverrides none() { return new PromptOverrides(null, Map.of(), Map.of()); }
}

File: server/csharp/MetaObjects.Render/Prompt/PromptOverrides.cs
namespace MetaObjects.Render.Prompt;

/// <summary>
/// Render-time overrides of the metadata defaults. <see cref="Style"/> <c>null</c> keeps the spec's
/// style; the maps override <see cref="PromptField.Example"/>/<see cref="PromptField.Instruction"/>
/// per field name.
/// </summary>
public sealed record PromptOverrides
{
    public PromptStyle? Style { get; }
    public IReadOnlyDictionary<string, string> Examples { get; }
    public IReadOnlyDictionary<string, string> Instructions { get; }

    public PromptOverrides(
        PromptStyle? style,
        IReadOnlyDictionary<string, string>? examples,
        IReadOnlyDictionary<string, string>? instructions)
    {
        Style = style;
        Examples = examples is null
            ? EmptyMap
            : new Dictionary<string, string>(examples);
        Instructions = instructions is null
            ? EmptyMap
            : new Dictionary<string, string>(instructions);
    }

    private static readonly IReadOnlyDictionary<string, string> EmptyMap =
        new Dictionary<string, string>();

    /// <summary>No overrides — keep every metadata default.</summary>
    public static PromptOverrides None() => new(null, null, null);
}
